.github/workflows/test-package.yml

```diff
@@ -0,0 +1,57 @@
+name: Dart CI
+
+on:
+  # Run CI on pushes to the main branch, and on PRs against main.
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  PUB_ENVIRONMENT: bot.github
+
+jobs:
+  # Check code formatting and static analysis on a single OS (linux)
+  # against Dart dev.
+  analyze:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sdk: [dev]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dart-lang/setup-dart@v0.1
+        with:
+          channel: ${{ matrix.sdk }}
+      - name: Install dependencies
+        run: dart pub get
+      - name: Check formatting
+        run: dart format --output=none --set-exit-if-changed .
+      - name: Analyze code
+        run: dart analyze
+        if: ${{ always() && steps['Install dependencies'].outcome == 'success' }}
```

Comment:
You need to add an ID to the `Install dependencies` step and then do `steps.<id>.conclusion == 'success'` for this to work afaik :) (at least this is how we do it in package:mono_repo)

Reply:
You can also remove the `always() &&` afaik

Reply:
Ah thanks!

Reply:
Hmm, not sure about removing the `always()`.

> If your if expression does not contain any of the status functions it will automatically result with success().
> https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions

I read that as meaning that if I don't use a status function at all, it will be aborted if the previous one fails because it will assume success(). But maybe the if expression overrides that as well 🤷

Reply:
As far as I could tell when trying to figure this out for mono_repo, any `if` just completely overrides the default behavior (which is `success()`)

Reply:
Ok, I'll try removing it and then I'll upload a commit with a formatting error.

Reply:
We need the `always()` or the step won't run (discussed offline).